docs/src/testing/contracts-collection.md

```diff
@@ -1,24 +1,25 @@
-## How Contracts Are Collected
+# How Contracts Are Collected
 
-When you call `snforge test`, one of the things that `snforge` does is that it calls Scarb, particularly `scarb build`.
-It makes Scarb build all contracts from your package and save them to the `target/{current_profile}` directory
-(read more on [Scarb website](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html)).
+`snforge` supports two mechanism for collecting contracts used in tests.
+The default depends on Scarb version used and can be controlled with `--no-optimization` flag.
 
-Then, `snforge` loads compiled contracts from the package your tests are in, allowing you to declare the contracts in tests.
+- If using Scarb version greater or equal to
+  2.8.3, [optimized collection mechanism](contract-collection/new-mechanism.md) is used by default.
+- If using Scarb version below 2.8.3 or using `--no-optimization` flag with
+  `snforge test` [old collection mechanism](contract-collection/old-mechanism.md) is used.
 
-> ⚠️ **Warning**
+> 📝 **Note**
 >
-> Make sure to define `[[target.starknet-contract]]` section in your `Scarb.toml`, otherwise Scarb won't build your contracts.
-
-
-## Using External Contracts In Tests
-
-If you wish to use contracts from your dependencies inside your tests (e.g. an ERC20 token, an account contract),
-you must first make Scarb build them. You can do that by using `build-external-contracts` property in `Scarb.toml`, e.g.: 
-
-```toml
-[[target.starknet-contract]]
-build-external-contracts = ["openzeppelin::account::account::Account"]
-```
-
-For more information about `build-external-contracts`, see [Scarb documentation](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html#compiling-external-contracts).
+> When using Scarb versions older than 2.8.3 it is **not possible** to enable new mechanism.
+> Migrating to new Scarb version is required.
+
+## Differences Between Collection Mechanisms
+
+| Feature                                                 | Old Mechanism | Optimised Mechanism |
+|---------------------------------------------------------|---------------|---------------------|
+| Using contracts from `/src`                             | ✅             | ✅                   |
+| Using contracts from `/tests`                           | ❌             | ✅                   |
+| Using contracts from modules marked with `#[cfg(test)]` | ❌             | ✅                   |
+| Using contracts from dependencies                       | ✅             | ✅                   |
+| Contracts more closely resemble ones from real network  | ✅             | ❌                   |
+| Additional compilation step required (`scarb build`)    | ✅             | ❌                   |
```

Comment:
With the current version it kind of suggests that having this additional compilation step is better. Wouldn't it be more appropriate to rename it to `Less compilation steps required`? Then the green mark would be on the optimized side.

Reply:
Hmm, I get your point, but maybe have it as separate points, close to each other? I think it is worth mentioning that using `scarb build --test` to build contracts can make them not 100% like the ones deployed to the network.

Though maybe it can be at least reworded to "Contracts can contain code that is not deployable on real network"?

Reply:
>I think it is worth mentioning that using scarb build --test to build contracts can make them not 100% like the ones deployed to the network.

Agree, but this is covered by the point `Contracts more closely resemble ones from real network `

> "Contracts can contain code that is not deployable on real network"

This sentence is confusing in my opinion